pkg/daemon/ceph/cleanup/radosnamespace.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleanup
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+)
+
+func RadosNamespaceCleanup(context *clusterd.Context, clusterInfo *client.ClusterInfo, poolName, radosNamespace string) error {
+	logger.Infof("starting clean up of CephBlockPoolRadosNamespace %q resources in blockpool %q", radosNamespace, poolName)
+
+	images, err := cephclient.ListImages(context, clusterInfo, poolName, radosNamespace)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list images in blockpool %q in rados namespace %q", poolName, radosNamespace)
+	}
+
+	var retErr error
+	for _, image := range images {
+		err := cephclient.PurgeImageSnapshots(context, clusterInfo, poolName, image.Name, radosNamespace)
+		if err != nil {
+			retErr = errors.Wrapf(err, "failed to purges snapshots of the image %q in pool %q.", image.Name, poolName)
+			logger.Error(retErr)
+		}
+
+		// TODO: rather than deleting add tasks to trash the image?
```

Comment:
What does this mean to add tasks to trash it? Since force deleting, seems like we should directly delete all the images as you have here.

Reply:
Removed the TODO for now. 
There was one suggestion by Madhu to use following command instead of deleting the images
``` 
rbd trash mv <RBD image image>
ceph rbd task trash remove <RBD image>
```